plugins/nf-amazon/src/main/com/upplication/s3fs/S3OutputStream.java

```diff
@@ -641,14 +646,11 @@ int getPartsCount() {
      */
     static synchronized ExecutorService getOrCreateExecutor(int maxThreads) {
         if( executorSingleton == null ) {
-            ThreadPoolExecutor pool = new ThreadPoolExecutor(
+            ThreadPoolExecutor pool = ThreadPoolBuilder.io(
```

Comment:
Using the thread pool configuration in this PR? Can you please make another try with the one I'm suggesting? 

Think we should also important running into a container using 2 GB of memory, and setting the JVM like 

```
-XX:InitialRAMPercentage=30 -XX:MaxRAMPercentage=70
```

Reply:
I've been running several benchmarks but I think we can't set the pool size so huge as 10_000

If we configure the executor with a huge pool the memory will be exhausted because the `write` method will be faster than the `upload` and we'll be requesting new ByteBuffers instead to reuse them.

Right now I'm instantiating the ThreadPoolBuilder with following logic: 

`ThreadPoolBuilder.io(1,maxThreads*3, maxThreads*10,"S3OutputStream")`  (i.e. a configurable pool based in threads)

With this configuration, I'm able to upload a 100GB file from **a container with 2GB of memory** in 40 mnts

- container with **2gb** memory and JAVA_OPTS="-XX:InitialRAMPercentage=30 -XX:MaxRAMPercentage=70"

file 50gb | 1 thread | chuncksize 30 mb | 25 mnts
file 100gb | 1 thread | chuncksize 30 mb | 41 mnts

- container with **3gb** memory and JAVA_OPTS="-XX:InitialRAMPercentage=30 -XX:MaxRAMPercentage=70"

file 100gb | 2 thread | chuncksize 20 mb | 41 mnts

Reply:
> If we configure the executor with a huge pool the memory will be exhausted because the write method will be faster than the upload and we'll be requesting new ByteBuffers instead to reuse them.

Good point! ok, pick the setting you found better